test/e2e/step-definitions/journal-steps.ts

```diff
@@ -63,6 +63,48 @@ When('I rekey a test for {string}', (candidateName) => {
   clickElement(buttonElement);
 });
 
+When(/^I start the test (early|late) for \"(.+)\"$/, (testTime: string, candidateName: string) => {
+  if (testTime === 'early') {
+    startingEarlyTest(candidateName);
+
+    // If the start test early dialog is shown just select continue
+    const startTestEarlyButton = element(by.id('early-start-start-test-button'));
+    startTestEarlyButton.isPresent().then((result) => {
+      if (result) {
+        clickElement(startTestEarlyButton);
+      }
+    });
+  }
+
+  if (testTime === 'late') {
+    startingExpiredTest(candidateName);
+
+    // If the rekey dialog is shown so just select start test normally
+    const lateStartTestButton = element(by.id('rekey-start-test-button'));
+    lateStartTestButton.isPresent().then((result) => {
+      if (result) {
+        clickElement(lateStartTestButton);
+      }
+    });
+  }
+});
+
+When('I rekey a late test for {string}',(candidateName) => {
+  const buttonElement = getElement(by.xpath(`//button/span/h3[text()[normalize-space(.) = "Start test"]]
+    [ancestor::ion-row/ion-col/ion-grid/ion-row/ion-col/candidate-link/div/button/span/
+    h3[text() = "${candidateName}"]]`));
```

Comment:
This feels like it can break really easily. Line 94 could be `[ancestor::candidate-link/div/button/span/ ' and that should work

Reply:
The is what is used through out the code to press the start button.

Reply:
The work around you have given me does not work and I was not able to find a way to make it shorter.